scarlet/psf.py

```diff
@@ -1,72 +1,100 @@
 import autograd.numpy as np
 import autograd.scipy as scipy
 from .bbox import Box
-from .component import Component
+from .model import Model, abstractmethod
 from .parameter import Parameter
 from .fft import Fourier
 from .observation import Observation
 
 
-def moffat(y, x, alpha=4.7, beta=1.5, bbox=None):
-    """Symmetric 2D Moffat function
+class PSF(Model):
+    @abstractmethod
+    def get_model(self, *parameter, offset=None):
```

Comment:
For the PSF, `offset` is a condition of the realization, not a parameter of the model. For example, a `GaussianPSF` has a width as `Parameter`, but the center of the PSF is by construction in the middle of the box. The `ImagePSF` from LSST have the same construction. 

On the other hand, a `PointSource` has a center parameter, but it's not shared by other PointSources, and certainly not by the underlying `PSF`.

Reply:
It's been a while, but I thought that part of the idea was to have the ability to allow the offset of the PSF to shift slightly in order to make up for misalignments in the PSF between bands. Did I misunderstand something or am I just making that up?

Reply:
No, you are right about that, but I'm thinking of a different mechanism for that. A PSF mismatch would be an `Observation`-level parameter, i.e. `Observation` gets another parameter `offset`, which it exposes to `Blend` through its `parameter` property. So, the PSF stays as it is (because offset isn't a PSF parameter unlike e.g. `sigma`), instead the observational frame gets offset. I'm not sure if we need this to be band-dependent, but I would rather define the PSF model to be centered, and treat the offset as part of the WCS.

This is different from actually allowing the PSF model itself to change, for instance by adjusting `sigma` of `GaussianPSF`. That would be a PSF parameter, exposed through `Observation.parameter`.

In both cases, we still need to ability to recompute the matching kernels. Next PR...

Reply:
So `offset` is here in case there is a PSF that we know is not centered in the box? Do we have a use case for this?

Reply:
I just used to produce a makeshift image of a star.

Reply:
And in general you will want to use it without having to specify sigma and instead use the current model.